webrtc/RTCPeerConnection-iceConnectionState.https.html

```diff
@@ -371,4 +371,30 @@
   assert_array_equals(pc2.iceStates, ['new', 'checking', 'connected']);
 }, 'Responder ICE connection state behaves as expected');
 
+/*
+  Test case for step 11 of PeerConnection.close().
+  ...
+  11. Set connection's ICE connection state to "closed". This does not invoke
+      the "update the ICE connection state" procedure, and does not fire any
+      event.
+  ...
+*/
+promise_test(async t => {
+  const pc1 = new RTCPeerConnection();
+  t.add_cleanup(() => pc1.close());
+  const pc2 = new RTCPeerConnection();
+  const stream = await getNoiseStream({ audio: true });
+  t.add_cleanup(() => stream.getTracks().forEach(track => track.stop()));
+
+  stream.getTracks().forEach(track => pc1.addTrack(track, stream));
+  exchangeIceCandidates(pc1, pc2);
+  doSignalingHandshake(pc1, pc2);
+  await listenToIceConnected(pc2);
+
+  pc2.oniceconnectionstatechange = t.unreached_func();
+  pc2.close();
+  await Promise.resolve();
+  assert_true(pc2.iceConnectionState === 'closed');
```

Comment:
Actually, we want to assert *first*, to test the state is closed synchronously, followed by allowing sufficient time to ensure an event does not come in.

Unfortunately, `await Promise.resolve()` only queues a microtask, which merely defers execution to the end of *this* run of the main event loop. This is not enough time for an event to have fired, since the firing of this event would be queued in a new task. We need:
```js
await new Promise(r => t.step_timeout(r));
```

Reply:
Yes, ICE connection state should be set to "closed" synchronously. I'm going to submit a new PR to fix it and wait 100ms for `oniceconnectionstatechange` event.